### PR TITLE
refactor config to work with load and save

### PR DESCRIPTION
**Pull request by joshuabrink1**
- refactor save and load functions to be used for config parsing
- include examples in `!help` command for `!save` and `!load`
- update cli to only require `--api_key` when there is no `--config` flag 
- update Readme to include config flag

### CODE REVIEW

File: src/revChatGPT/V3.py [hunk 1]
```diff
@@ -459,31 +420,28 @@ def main() -> NoReturn:
...
-        if args.api_key is None:
-            print(
-                "Add a config.v3.json and add the path using --config or add an api_key using --api_key",
-            )
-            # raising at top level is messy and can confuse some people
-            return
 
+    config = args.config or os.environ.get("GPT_CONFIG_PATH")
+    # Initialize chatbot
+    if config:
```
Comment (Arborym) [inline src/revChatGPT/V3.py:437]:
```suggestion
    # Initialize chatbot
    if config := args.config or os.environ.get("GPT_CONFIG_PATH"):
```

File: src/revChatGPT/V3.py [hunk 2]
```diff
@@ -459,31 +420,28 @@ def main() -> NoReturn:
...
+        chatbot = ChatbotCLI(args.api_key)
+        try:
+            chatbot.load(config)
+        except:
```
Comment (Arborym) [inline src/revChatGPT/V3.py:441]:
```suggestion
        except Exception:
```

File: src/revChatGPT/utils.py [hunk 3]
```diff
@@ -67,3 +67,27 @@ async def get_input_async(
...
+        given_keys = keys[1:]
+        return {key for key in class_keys if key not in given_keys}
+    else:
+        # Check if all passed keys are valid
+        if invalid_keys := set(keys) - class_keys:
+            raise ValueError(
+                f"Invalid keys: {invalid_keys}",
+            )
+        # Only return specified keys that are in class_keys
+        return {key for key in keys if key in class_keys}
```
Comment (Arborym) [inline src/revChatGPT/utils.py:93]:
```suggestion
        return {key for key in class_keys if key not in keys[1:]}
    # Check if all passed keys are valid
    if invalid_keys := set(keys) - class_keys:
        raise ValueError(
            f"Invalid keys: {invalid_keys}",
        )
    # Only return specified keys that are in class_keys
    return {key for key in keys if key in class_keys}
```

**Review by Arborym — CHANGES_REQUESTED** [review]
Few edits

**Review by acheong08 — APPROVED** [review]

**Pull request merged**